### PR TITLE
Poisson binomial distribution computations

### PR DESCRIPTION
**Pull request by dvadym**
Those computations will be used in partition selection utility analysis

### CODE REVIEW

File: utility_analysis_new/poisson_binomial.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+def compute_pmf_approximation(mean: float, sigma: float, skewness: float,
+                              n: int):
+    """Computes approximate probability mass function of poisson binomial distribution.
+
+    The computation is based on paper chapter 3.3 (refined normal approximation)
+    https://www.researchgate.net/publication/257017356_On_computing_the_distribution_function_for_the_Poisson_binomial_distribution
+    """
+    G = lambda x: (norm.cdf(x) + skewness * (1 - x * x) * norm.pdf(x) / 6)
+    xs = np.arange(-0.5, n + 1, 1)
```
Comment (sushkoy) [inline utility_analysis_new/poisson_binomial.py:53]: optional: wdyt on using the formula from the Chapter G((k + 0.5 − mean) / sigma) with 
k = [0, 1, 2, ..]

File: utility_analysis_new/poisson_binomial.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    cdf_values = G((xs - mean) / sigma)
```
Comment (sushkoy) [inline utility_analysis_new/poisson_binomial.py:54]: Do we need to clamp CDF values to [0, 1] as the article suggests?

**Review by sushkoy — APPROVED** [review]

File: utility_analysis_new/poisson_binomial.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+def compute_exp_std_skewness(ps: Sequence[float]) -> Tuple[float, float, float]:
+    exp = np.sum(ps)
+    std = np.sqrt(np.sum([p * (1 - p) for p in ps]))
+    skewness = np.sum([p * (1 - p) * (1 - 2 * p) for p in ps]) / std**3
+    return exp, std, skewness
+
+
+def compute_pmf_approximation(mean: float, sigma: float, skewness: float,
```
[the remark this answers is not in the document]
Comment (dvadym) [reply utility_analysis_new/poisson_binomial.py:45]: No, mean, sigma, skewness need to be computed distributely, from data on many machines, we can't assume that probabilities will always fit in memory of one machine, so we need to compute mean, std, skewness with combiners in scalable way.

File: utility_analysis_new/poisson_binomial.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+                              n: int):
+    """Computes approximate probability mass function of poisson binomial distribution.
```
[the remark this answers is not in the document]
Comment (dvadym) [reply utility_analysis_new/poisson_binomial.py]: No :(

File: utility_analysis_new/poisson_binomial.py (hunk 1 above)
Comment (dvadym) [reply utility_analysis_new/poisson_binomial.py]: Good point, done!

File: utility_analysis_new/poisson_binomial.py (hunk 2 above)
Comment (dvadym) [reply utility_analysis_new/poisson_binomial.py]: Thanks, yeah it's better to clip

**Review by dvadym — COMMENTED** [review]
Thanks for very useful comments!

**Pull request merged**